custom/icds_reports/templates/icds_reports/icds_app/mobile/awc-reports.html

```diff
@@ -3,21 +3,50 @@
   <div class="app-bg h-100vh overflow-hidden" ng-show="!$ctrl.filtersOpen && $ctrl.selectedLocationLevel==4">
     <div class="top-section p-3" style="padding-bottom: 25vh">
       {% include 'icds_reports/icds_app/mobile/partials/mobile-heading.html' %}
-      <div class="d-flex no-gutters align-items-center justify-content-around mb-5">
-          <div class="position-relative" ng-repeat="step in $ctrl.mobileSteps">
+      <div class="d-flex no-gutters align-items-center mb-5 overflow-auto justify-content-between">
+          <div class="position-relative m-3" ng-repeat="step in $ctrl.mobileSteps">
               <dot-link id="{$ step.id $}" label="{$ step.label $}" route="{$ step.route $}" image="{$ step.image $}"></dot-link>
           </div>
       </div>
-      <p class="font-weight-700 text-white font-16 font-montserrat" ng-bind="$ctrl.currentStepMeta.label"></p>
+      <p class="font-weight-700 text-white font-16 font-montserrat" ng-bind="$ctrl.currentStepMeta.label" ng-if="!$ctrl.isTabularDataDisplayed"></p>
+      <div class="d-flex justify-content-between align-items-start px-3" ng-if="$ctrl.isTabularDataDisplayed">
+          <p class="font-weight-700 text-white font-16 font-montserrat w-65">{$ $ctrl.currentStepMeta.label $}</p>
+          <div class="w-35">
```

Comment:
is it not possible to extract this to its own component?

Reply:
I will check for possibility